xarray/backends/api.py

```diff
@@ -15,35 +15,53 @@
 from ..core.combine import auto_combine
 from ..core.utils import close_on_error, is_remote_uri
 from ..core.pycompat import basestring, path_type
+from ..core.options import OPTIONS
 
 DATAARRAY_NAME = '__xarray_dataarray_name__'
 DATAARRAY_VARIABLE = '__xarray_dataarray_variable__'
 
 
+def _guess_file_format(path):
```

Comment:
A unit test for this function would be quite valuable (and would be pretty easy to write).